mixnet/node/src/config.rs

```diff
@@ -0,0 +1,25 @@
+use std::net::{Ipv4Addr, SocketAddr, SocketAddrV4};
+
+use nym_sphinx::PrivateKey;
+use serde::{Deserialize, Serialize};
+use sphinx_packet::crypto::PRIVATE_KEY_SIZE;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct MixnetNodeConfig {
+    // A listen address for receiving Sphinx packets
+    pub listen_address: SocketAddr,
+    // A key for decrypting Sphinx packets
+    pub private_key: [u8; PRIVATE_KEY_SIZE],
+    // An address of the client with receiver-mode enabled, for sending final payloads
+    pub client_address: SocketAddr,
```

Comment:
@Zeegomo @danielSanchezQ I just added 80e49f2 that changes the design a bit. Sorry to put this commit in this PR, but I think it's even easier to review.

In the previous design, the problem was that message senders must specify the mixnet destination (== the IP of `MixnetClient`).
Please note that it's different from the message routing destination).
But, in reality, senders doesn't have any information about the IP of `MixnetClient`s available.

Instead, I refined the design like this:
- A `nomos-node` always embeds a `MixnetClient` that just receives Sphinx packets, reconstructs messages, and sends them to its network backend.
   - Actually, this is the same in the previous design.
- If you want to run a `MixnetNode`, you must know at least one `MixnetClient` address.
  - In other words, `MixnetNode` operators should run their own `nomos-node` as well.
- When message senders choose 3 `MixnetNode` from `MixnetTopology`, they choose one more `MixnetNode` as a destination, randomly.
- Finallly, all message fragments (sent from a message sender) are forwarded to the destination `MixnetNode`. Then, the message is reconstructed and is broadcasted/routed to all other `nomos-node`s.
- On the other hand, as a `nomos-node` operator, it's not mandatory to run `MixnetNode`.

<img width="1317" alt="image" src="https://github.com/logos-co/nomos-node/assets/5462944/f6cbbaf3-3831-4591-8128-033eff74499e">

With this design, the message sender doesn't need to know about the mixnet destination.